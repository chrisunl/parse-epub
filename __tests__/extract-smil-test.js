import { contentOpf, testSmil } from './data.js';
import extractManifest from '../extract/manifest.js';
import extractMetadata from '../extract/metadata.js';
import extractSmil from '../extract/smil.js';

const SMIL_ID = 's0036';

test('#extractSmil', () => {
  const metadata = extractMetadata(contentOpf, extractManifest(contentOpf));
  expect(extractSmil(testSmil, SMIL_ID, metadata.mediaOverlayDurations, './OPS')).toMatchSnapshot();
});