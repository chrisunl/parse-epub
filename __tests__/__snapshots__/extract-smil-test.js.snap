exports[`test #extractSmil 1`] = `
Object {
  "body": Object {
    "childNodes": Array [
      Object {
        "childNodes": Array [
          Object {
            "audio": Object {
              "clipBegin": 0,
              "clipEnd": 3.69,
              "id": undefined,
              "src": "/audio/sn_77e0-c01-s02-020.mp3",
            },
            "id": undefined,
            "isPar": true,
            "text": Object {
              "id": undefined,
              "src": "/s9ml/culture_and_belonging/selena_s_blog.xhtml#data-uuid-483ef689101641c8a9b8c1b32c707d19",
              "srcFile": "/s9ml/culture_and_belonging/selena_s_blog.xhtml",
              "srcFragmentId": "data-uuid-483ef689101641c8a9b8c1b32c707d19",
            },
            "textref": undefined,
            "type": undefined,
          },
          Object {
            "audio": Object {
              "clipBegin": 0,
              "clipEnd": 25.089,
              "id": undefined,
              "src": "/audio/sn_77e0-c01-s03-023.mp3",
            },
            "id": undefined,
            "isPar": true,
            "text": Object {
              "id": undefined,
              "src": "/s9ml/culture_and_belonging/selena_s_blog.xhtml#data-uuid-2e2efa6ce8074352a82060a20b9294a9",
              "srcFile": "/s9ml/culture_and_belonging/selena_s_blog.xhtml",
              "srcFragmentId": "data-uuid-2e2efa6ce8074352a82060a20b9294a9",
            },
            "textref": undefined,
            "type": undefined,
          },
          Object {
            "audio": Object {
              "clipBegin": 0,
              "clipEnd": 39.859,
              "id": undefined,
              "src": "/audio/sn_77e0-c01-s03-024.mp3",
            },
            "id": undefined,
            "isPar": true,
            "text": Object {
              "id": undefined,
              "src": "/s9ml/culture_and_belonging/selena_s_blog.xhtml#data-uuid-618ee9678b494b278e0592ede884e361",
              "srcFile": "/s9ml/culture_and_belonging/selena_s_blog.xhtml",
              "srcFragmentId": "data-uuid-618ee9678b494b278e0592ede884e361",
            },
            "textref": undefined,
            "type": undefined,
          },
          Object {
            "audio": Object {
              "clipBegin": 0,
              "clipEnd": 28.289,
              "id": undefined,
              "src": "/audio/sn_77e0-c01-s03-026.mp3",
            },
            "id": undefined,
            "isPar": true,
            "text": Object {
              "id": undefined,
              "src": "/s9ml/culture_and_belonging/selena_s_blog.xhtml#data-uuid-333ff0a0742c4d25b4e6c2fb5ac51764",
              "srcFile": "/s9ml/culture_and_belonging/selena_s_blog.xhtml",
              "srcFragmentId": "data-uuid-333ff0a0742c4d25b4e6c2fb5ac51764",
            },
            "textref": undefined,
            "type": undefined,
          },
          Object {
            "audio": Object {
              "clipBegin": 0,
              "clipEnd": 31.82,
              "id": undefined,
              "src": "/audio/sn_77e0-c01-s03-027.mp3",
            },
            "id": undefined,
            "isPar": true,
            "text": Object {
              "id": undefined,
              "src": "/s9ml/culture_and_belonging/selena_s_blog.xhtml#data-uuid-2ba50ef8103a4961807dde37efd4664d",
              "srcFile": "/s9ml/culture_and_belonging/selena_s_blog.xhtml",
              "srcFragmentId": "data-uuid-2ba50ef8103a4961807dde37efd4664d",
            },
            "textref": undefined,
            "type": undefined,
          },
          Object {
            "audio": Object {
              "clipBegin": 0,
              "clipEnd": 21.55,
              "id": undefined,
              "src": "/audio/sn_77e0-c01-s03-028.mp3",
            },
            "id": undefined,
            "isPar": true,
            "text": Object {
              "id": undefined,
              "src": "/s9ml/culture_and_belonging/selena_s_blog.xhtml#data-uuid-154cfc4c605546708794e73f419336e0",
              "srcFile": "/s9ml/culture_and_belonging/selena_s_blog.xhtml",
              "srcFragmentId": "data-uuid-154cfc4c605546708794e73f419336e0",
            },
            "textref": undefined,
            "type": undefined,
          },
          Object {
            "audio": Object {
              "clipBegin": 0,
              "clipEnd": 22.25,
              "id": undefined,
              "src": "/audio/sn_77e0-c01-s03-029.mp3",
            },
            "id": undefined,
            "isPar": true,
            "text": Object {
              "id": undefined,
              "src": "/s9ml/culture_and_belonging/selena_s_blog.xhtml#data-uuid-c7e93d0d9d8b4159a8f1b2517ed9a0ac",
              "srcFile": "/s9ml/culture_and_belonging/selena_s_blog.xhtml",
              "srcFragmentId": "data-uuid-c7e93d0d9d8b4159a8f1b2517ed9a0ac",
            },
            "textref": undefined,
            "type": undefined,
          },
          Object {
            "audio": Object {
              "clipBegin": 0,
              "clipEnd": 33.109,
              "id": undefined,
              "src": "/audio/sn_77e0-c01-s03-030.mp3",
            },
            "id": undefined,
            "isPar": true,
            "text": Object {
              "id": undefined,
              "src": "/s9ml/culture_and_belonging/selena_s_blog.xhtml#data-uuid-f4a776b35fb54a0db8f7c12135b7b1b8",
              "srcFile": "/s9ml/culture_and_belonging/selena_s_blog.xhtml",
              "srcFragmentId": "data-uuid-f4a776b35fb54a0db8f7c12135b7b1b8",
            },
            "textref": undefined,
            "type": undefined,
          },
        ],
        "id": undefined,
        "isPar": false,
        "textref": "selena_s_blog.xhtml",
        "type": undefined,
      },
    ],
    "id": undefined,
    "textref": undefined,
    "type": undefined,
  },
  "id": "s0036",
  "version": "3.0",
}
`;
